pkg/blobcache/blobcache.go

```
@@ -0,0 +1,410 @@
+	return rc, size, err
+}
+
+func (s *blobCacheSource) GetSignatures(ctx context.Context, instanceDigest *digest.Digest) ([][]byte, error) {
+	return s.source.GetSignatures(ctx, instanceDigest)
+}
+
+func (s *blobCacheSource) LayerInfosForCopy(ctx context.Context) ([]types.BlobInfo, error) {
+	return nil, nil
+}
+
+func (d *blobCacheDestination) Reference() types.ImageReference {
+	return d.reference
+}
+
+func (d *blobCacheDestination) Close() error {
+	return d.destination.Close()
+}
+
+func (d *blobCacheDestination) SupportedManifestMIMETypes() []string {
+	return d.destination.SupportedManifestMIMETypes()
+}
+
+func (d *blobCacheDestination) SupportsSignatures(ctx context.Context) error {
+	return d.destination.SupportsSignatures(ctx)
+}
+
+func (d *blobCacheDestination) DesiredLayerCompression() types.LayerCompression {
+	return d.destination.DesiredLayerCompression()
+}
+
+func (d *blobCacheDestination) AcceptsForeignLayerURLs() bool {
+	return d.destination.AcceptsForeignLayerURLs()
+}
+
+func (d *blobCacheDestination) MustMatchRuntimeOS() bool {
+	return d.destination.MustMatchRuntimeOS()
+}
+
+func (d *blobCacheDestination) IgnoresEmbeddedDockerReference() bool {
+	return d.destination.IgnoresEmbeddedDockerReference()
+}
+
+func (d *blobCacheDestination) PutBlob(ctx context.Context, stream io.Reader, inputInfo types.BlobInfo, isConfig bool) (types.BlobInfo, error) {
+	if inputInfo.Digest != "" && len(d.reference.directories) > 0 {
+		if d.reference.removeFirstDirectoryOnCleanup && d.reference.directories[0] == "" {
+			temp, err := ioutil.TempDir("", "")
+			if err != nil {
+				return types.BlobInfo{}, errors.Wrapf(err, "error creating new temporary directory for blob cache")
+			}
+			d.reference.directories[0] = temp
+		}
+		filename := filepath.Join(d.reference.directories[0], makeFilename(inputInfo.Digest, isConfig))
+		f, err := os.OpenFile(filename, os.O_CREATE|os.O_TRUNC|os.O_WRONLY, 0600)
+		if err == nil {
```

Are you ignoring err != nil?
Should we only ignore ENOEXIST?

---

Yes, and probably yes.  Adding a debug message for the `!os.IsNotExist()` case.